EIPS/eip-7432.md

```diff
@@ -0,0 +1,346 @@
+      "description": "Describes the asset to which this NFT represents"
+    },
+    "image": {
+      "type": "string",
+      "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive"
+    }
+  },
+  
+  /** Additional fields for Roles **/
+
+  "roles": [
+    {
+      "id": {
+        "type": "bytes32",
+        "description": "Identifies the role"
+      },
+      "name": {
+        "type": "string",
+        "description": "Human-readable name of the role"
+      },
+      "description": {
+        "type": "string",
+        "description": "Describes the role"
+      },
+      "supportsMultipleAssignments": {
+        "type": "boolean",
+        "description": "Whether the role supports simultaneous assignments or not"
+      },
+      "inputs": [
+        {
+          "name": {
+            "type": "string",
+            "description": "Human-readable name of the argument"
+          },
+          "type": {
+            "type": "string",
+            "description": "Solidity type, e.g., uint256 or address"
+          }
+        }
+      ]
+    }
+  ]
+  
+}
+```
+
+The following JSON is an example of [ERC-7432](./eip-7432.md) Metadata:
+
+```js
+{
+  // ... Existing NFT Metadata
+  
+  "roles": [
+    {
+      // keccak256("PROPERTY_MANAGER")
+      "id": "0x5cefc88e2d50f91b66109b6bb76803f11168ca3d1cee10cbafe864e4749970c7",
+      "name": "Property Manager",
+      "description": "The manager of the property is responsible for furnishing it and ensuring its good condition.",
+      "supportsMultipleAssignments": false,
+      "inputs": []
+    },
+    {
+      // keccak256("PROPERTY_TENANT")
+      "id": "0x06a3b33b0a800805559ee9c64f55afd8a43a05f8472feb6f6b77484ff5ac9c26",
+      "name": "Property Tenant",
+      "description": "The tenant of the property is responsible for paying the rent and keeping the property in good condition.",
+      "supportsMultipleAssignments": true,
+      "inputs": [
+        {
+          "name": "rent",
+          "type": "uint256"
+        }
+      ]
+    }
+  ]
+  
+}
+```
+
+The `roles` array properties are SUGGESTED, and developers should add any other relevant information as necessary (e.g.,
+an image for the role). However, it's highly RECOMMENDED to include the `supportsMultipleAssignments` property, as this
+field is required in the `hasRole` function (refer back to [Unique and Non-Unique Roles](#unique-and-non-unique-roles)).
+
+It's also important to highlight the importance of the `inputs` property. This field describes the parameters that
+should be encoded and passed to the `grantRole` function. It's RECOMMENDED to use the properties `type` and `components`
+defined on the Solidity ABI Specification, where `type` is the canonical type of the parameter, and `components` is used
+for complex tuple types.
```

Comment:
This is also specification, so it should go into that section.

Reply:
Moved!